cmd/config/set.go

```diff
@@ -221,24 +246,26 @@ func configSetContext(cmd *cobra.Command, args []string) {
 		}
 		ctxPtr.CsvFile = "" // CSV file is a backward-compatibility value only
 	}
-	if flags.Changed("auth") {
-		val, _ := flags.GetString("auth")
-		if val != "" && !slices.Contains(GetAuthMethodsStringList(), val) {
-			log.Fatalf(`Invalid --auth method %q; must be one of {"%v"}`, val, strings.Join(GetAuthMethodsStringList(), `", "`))
-		}
-		ctxPtr.AuthMethod = val
-	}
 
 	if ctxPtr.AuthMethod == AuthMethodLocal {
 		if flags.Changed(AppdPid) {
+			if ctxPtr.AuthMethod != "service-principal" && ctxPtr.AuthMethod != "agent-principal" && !flags.Changed("patch") {
```

Comment:
unnecessary/confusing check. We already know we're in AuthMethodLocal auth mode (see line 250 above)